python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,730 @@
+            )
+
+    def _init_ntype_maps(self):
+        """Initialization function to build unique id maps for node types of
+        the input graph. This function creates a list of node types present in
+        the graph, and unique-id <-> node type and reverse maps.
+        """
+        self._ntypes = self._data[STR_NODE_TYPE]
+        self._ntype_id_map = {
+            ntype_name: idx for idx, ntype_name in enumerate(self._ntypes)
+        }
+        self._id_ntype_map = dict(enumerate(self._ntypes))
+
+    def _init_ntype_features(self):
+        """Initilization function to build a map in which keys are node types
+        in the graph and values are a lists of features for the corresponding
+        node type, if present. For instance,
+            ntype1: [feature1, feature2, ...] where,
+            ntype1 is a node type from the graph and
+            [feature1, feature2, ....] is a list of features associated with
+            the ntype1 node type.
+        """
+        self._ntype_features = {}
+        node_data = self._data.get(STR_NODE_DATA, {})
+        for ntype in self._data[STR_NODE_TYPE]:
+            features = node_data.get(ntype, {})
+            if len(features) > 0:
+                self._ntype_features[ntype] = list(features.keys())
+
+    def _init_ntype_feature_files(self):
+        """Initialization function to build a map to capture the files names
+        and their format. For instance,
+            (ntype, feature_name): (file_type, delimiter, [file1, file2, ...]
+            where ``ntype`` is the node type,
+                  ``feature_name`` is a feature associated with ``ntype`` node
+                  type,
+                  ``file_type`` is the type of the file numpy and parquet are
+                  supported file formats,
+                  ``delimiter`` is the delimiter used when these set of files
+                  are created, and
+                  ``[file1, file2, ...]`` is a list of files in which features
+                  are stored for this node type.
+        """
+        self._ntype_feature_files = {}
+        node_data = self._data.get(STR_NODE_DATA, {})
+        for ntype, ntype_info in node_data.items():
+            for feature_name, feature_info in ntype_info.items():
+                data_files = feature_info.get(STR_DATA, [])
+                file_type = None
+                delimiter = None
+                if len(data_files) > 0:
+                    file_type = feature_info[STR_FORMAT][STR_NAME]
+                    delimiter = feature_info[STR_FORMAT].get(
+                        STR_FORMAT_DELIMITER, ","
+                    )
+
+                self.__check_ifexists(data_files, f"{ntype}/{feature_name}")
+                self._ntype_feature_files[(ntype, feature_name)] = (
+                    file_type,
+                    delimiter,
+                    data_files,
+                )
+
+    ############################## Public Methods ##############################
```

Comment:
why do you use methods instead of properties?

Reply:
Python does not allow properties with parameters.

Reply:
since you already return them as a property, why cannot you just do something like
```
schema.etype_info[etype]
```